Spec sample update

lgolding opened the pull request:
Update checked-in samples per Annex G of the spec.

michaelcfanning commented on src/Sarif.FunctionalTests/SpecExamples/Comprehensive.sarif:
```diff
@@ -28,20 +28,18 @@
         "endTime": "2016-07-16T14:19:01Z",
         "machine": "BLD01",
-        "account": "builder",
+        "account": "build",
```
how about buildAgent or some value that suggests an actual account?

lgolding replied:
Done.

---
In reply to: [72279460](https://github.com/Microsoft/sarif-sdk/pull/464#discussion_r72279460) [](ancestors = 72279460)

michaelcfanning commented on src/Sarif.FunctionalTests/SpecExamples/Comprehensive.sarif:
```diff
@@ -74,42 +82,153 @@
             "formatId": "default",
             "arguments": [
-              "n"
+              "p"
```
is there an expanded argument value over a single letter that might be more illustrative?

lgolding replied:
Sure, your suggestion of `ptr` below is fine.

---
In reply to: [72279756](https://github.com/Microsoft/sarif-sdk/pull/464#discussion_r72279756) [](ancestors = 72279756)

michaelcfanning commented on src/Sarif.FunctionalTests/SpecExamples/Comprehensive.sarif:
```diff
@@ -74,42 +82,153 @@
+          ],
+          "relatedLocations": [
+            {
+              "message": "\"count\" was declared here.",
```
we should add optional formattedMessage to ACL. imo.

lgolding replied:
We've discussed this. The reason we didn't was that we didn't know where to put the message formats (it wasn't obvious that they belonged in rule.messageFormats, since they weren't necessarily rule-specific. Until we have a use case, I'd wait on this.

---
In reply to: [72280081](https://github.com/Microsoft/sarif-sdk/pull/464#discussion_r72280081) [](ancestors = 72280081)

michaelcfanning replied:
Right. Would be great to have a tracking mechanism for this sort of thing. Wonder if it warrants a note in the specification. ?

lgolding replied:
I filed [https://github.com/sarif-standard/sarif-spec/issues/251](https://github.com/sarif-standard/sarif-spec/issues/251), "Should annotatedCodeLocation has a formattedMessage property?", and assigned it to Milestone=Future.

---
In reply to: [72288539](https://github.com/Microsoft/sarif-sdk/pull/464#discussion_r72288539) [](ancestors = 72288539)

michaelcfanning commented on src/Sarif.FunctionalTests/SpecExamples/Comprehensive.sarif:
```diff
@@ -74,42 +82,153 @@
                   "importance": "essential",
-                  "fullyQualifiedLogicalName": "collections::list::add",
+                  "message": "Variable \"p\" declared.",
+                  "snippet": "int *p;",
```
expand this name to 'ptr'. :)

lgolding replied:
Done.

---
In reply to: [72280197](https://github.com/Microsoft/sarif-sdk/pull/464#discussion_r72280197) [](ancestors = 72280197)

michaelcfanning commented on src/Sarif.FunctionalTests/SpecExamples/Comprehensive.sarif:
```diff
@@ -74,42 +82,153 @@
                   "kind": "declaration",
                   "importance": "essential",
-                  "fullyQualifiedLogicalName": "collections::list::add",
+                  "message": "Variable \"p\" declared.",
```
Can I suggest using single quotes for intra-message enquotation? We should recommend this in the spec in the formatting section. Makes the log file more readable.

lgolding replied:
I don't agree with this. My concern is less about making the log file readable and more about making the messages displayed to the end user be grammatical. There's nothing in the Microsoft Manual of Style that recommends single quotes, and Chicago is specific about double quotes except in certain cases (British English is one of them). I'm going to stick with double quotes unless you have a citation that specifies single quotes. (I know single quotes are widely used in MS error messages, probably for the reason you cite (readabilty of the message in code), but I've never agreed with it.)

---
In reply to: [72280385](https://github.com/Microsoft/sarif-sdk/pull/464#discussion_r72280385) [](ancestors = 72280385)

michaelcfanning replied:
We shouldn't ignore the weight of MS usage. One strong reason in our context (reporting messages) for favoring single quotes is that static analysis messages themselves are extracted and quoted in other contexts.

lgolding replied:
Let's discuss offline.

---
In reply to: [72287502](https://github.com/Microsoft/sarif-sdk/pull/464#discussion_r72287502) [](ancestors = 72287502)

michaelcfanning commented on src/Sarif.FunctionalTests/SpecExamples/Comprehensive.sarif:
```diff
@@ -74,42 +82,153 @@
+                  "step": 1,
+                  "kind": "assignment",
+                  "importance": "unimportant",
+                  "snippet": "offset = 0",
```
where's the trailing semicolon? :)

lgolding replied:
Fixed :)

---
In reply to: [72280460](https://github.com/Microsoft/sarif-sdk/pull/464#discussion_r72280460) [](ancestors = 72280460)

michaelcfanning commented on src/Sarif.FunctionalTests/SpecExamples/Comprehensive.sarif:
```diff
@@ -74,42 +82,153 @@
+                    }
+                  },
+                  "fullyQualifiedLogicalName": "collections::list:add",
+                  "module": "collections",
```
Is collections really the module here? We should try to distinguish it more from the namespace...

lgolding replied:
How about "utilities"?

---
In reply to: [72280747](https://github.com/Microsoft/sarif-sdk/pull/464#discussion_r72280747) [](ancestors = 72280747)

michaelcfanning replied:
Collections would be considered something more fundamental to the platform. How about 'platform', 'system', 'standardCollections', or some such?

lgolding replied:
ok, platform.

---
In reply to: [72287911](https://github.com/Microsoft/sarif-sdk/pull/464#discussion_r72287911) [](ancestors = 72287911)

lgolding replied:
Changed to "platform".

---
In reply to: [72286660](https://github.com/Microsoft/sarif-sdk/pull/464#discussion_r72286660) [](ancestors = 72286660,72280747)

lgolding replied:
Done.

---
In reply to: [72288126](https://github.com/Microsoft/sarif-sdk/pull/464#discussion_r72288126) [](ancestors = 72288126,72287911)

michaelcfanning commented on src/Sarif.FunctionalTests/SpecExamples/Comprehensive.sarif:
```diff
@@ -74,42 +82,153 @@
+              "message": "Call stack resulting from usage of uninitialized variable.",
+              "frames": [
+                {
+                  "message": "Exception thrown",
```
Don't we suggest adding tailing period for messages?

lgolding replied:
Yes, fixed.

---
In reply to: [72281289](https://github.com/Microsoft/sarif-sdk/pull/464#discussion_r72281289) [](ancestors = 72281289)

michaelcfanning commented on src/Sarif.FunctionalTests/SpecExamples/RuleMetadata.sarif:
```diff
@@ -3,15 +3,25 @@
+        "BA2006": {
+          "id": "BA2006",
+          "name": "BuildWithSecureTools",
+          "shortDescription": "Application code should be compiled with the most up-to-date tool sets ...",
```
What's with the fake ellipses in this content?

lgolding replied:
Because without them, `fullDescription` is the same as `shortDescription`. I just wanted to indicate that `fullDescription` would really be longer. I'll remove the ellipsis from `shortDescription` and from the `messageFormats` below. I'll ponder what to do about `fullDescription`.

---
In reply to: [72288078](https://github.com/Microsoft/sarif-sdk/pull/464#discussion_r72288078) [](ancestors = 72288078)

lgolding replied:
ok, fixed.

---
In reply to: [72288909](https://github.com/Microsoft/sarif-sdk/pull/464#discussion_r72288909) [](ancestors = 72288909,72288078)

michaelcfanning commented on src/Sarif.FunctionalTests/SpecExamples/RuleMetadata.sarif:
```diff
@@ -3,15 +3,25 @@
+          "id": "BA2006",
+          "name": "BuildWithSecureTools",
+          "shortDescription": "Application code should be compiled with the most up-to-date tool sets ...",
+          "fullDescription": "Application code should be compiled with the most up-to-date tool sets ...",
```
I believe toolset may be an accepted aggregation (agglutination? compoundination) these days.

lgolding replied:
I got this text straight from BinSkim. And even if "toolset" is acceptable, surely "tool set" is also acceptable.

---
In reply to: [72288198](https://github.com/Microsoft/sarif-sdk/pull/464#discussion_r72288198) [](ancestors = 72288198)

michaelcfanning commented on src/Sarif.FunctionalTests/SpecExamples/RuleMetadata.sarif:
```diff
@@ -3,15 +3,25 @@
+          "messageFormats": {
+            "Error_BadModule": "built with {0} compiler version {1} (Front end version {2})",
+            "Pass": "{0}' was built with tools that satisfy configured policy.",
+            "Error": "{0}' was compiled with one or tools that do not satisfy configured policy ...",
```
Again, since this text looks fine (complete sentence) why provide apparent continuation markers?

lgolding commented:
@michaelcfanning I've addressed all your feedback. The only remaining issue is the single/double quote question, which we're discussing offline. So I'm going to merge this, and copy the updated files back into the spec PR, sarif-standard/sarif-spec#248.

michaelcfanning commented:
terrific +1
